packages/dev/core/src/Maths/math.color.ts

```diff
@@ -29,7 +30,9 @@ function colorChannelToGammaSpaceExact(color: number): number {
 /**
  * Class used to hold a RGB color
  */
-export class Color3 {
+export class Color3 implements Tensor<[number, number, number]> {
+    public readonly dimension: [number] = [3];

```

Comment:
This would compile to instance property in typescript, this could be directly attached to prototype to save memory.
```typescript

class Color3  {
    declare public readonly dimension: Readonly<[number]>;

}

Object.defineProperty(Color3.prototype, 'dimension', {
    value: [3],
});

```

See also the typescript [playground](https://www.typescriptlang.org/play?#code/FAYwNghgzlAEDCB7MiBOBmA8qglgcxwDsIxYyBvYMsgBwFcAjMHEWVAUwgBNFCwBPWFxwBbdoSg5eALlgBtQnREN2qALqwAvPPRqA3MGABfUJBgJkadBSrUu7cBA6x6TFm048+g4WIlTCWQAlT14BAB4FJRV1AD4DY0NMBgArBwAXADp7ADMidgAFVEQaVXT+AAokFAxMmmL0xHLSgBpYAHJfcUledrbKalgANxI6dlk5XRbjAEoEkF4oZHZMlDwKwnYAdwsa9AqZ7NFugJmgA) for more details.

Reply:
I would expect TS to compile correctly since it is `readonly`. If not, I wouldn't expect the performance difference to be noticeable.

Reply:
> I wouldn't expect the performance difference to be noticeable, but if it is then your approach would help.

In [Animation](https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Animations/animation.ts#L1029C35-L1029C35) for example, animations loaded from ~10MB [glb model](https://github.com/donmccurdy/glTF-Transform/files/8081046/Archive.zip) could contain ~700k keyframes, each one of which can contain one Vector3 or Quaternion, so this memory affect on every instance should be non-trivial.

![example](https://github.com/BabylonJS/Babylon.js/assets/120431159/80b44b1b-2dd8-447d-94a7-9808ebdae817)

Here are memory usage mesured in read world using this Arm.glb model:
![profiling](https://github.com/BabylonJS/Babylon.js/assets/120431159/54232157-3436-40c4-9d9b-bdaa9e88f815)
The current release takes 107MB total, while the pr takes 116MB.